### PR TITLE
STORY-18969 simplify quoting enums code

### PR DESCRIPTION
**Pull request by ttstarck**
`ActiveRecord::Base.send(:quote_bound_value, value.to_s)` is just calling `quote(value)` when passing a string, and this fixes issues we are seeing in our ActiveTableSet gem. The issues stemmed from `ActiveRecord::Base#quote_bound_value` using a different connection

PR in web showing 🟢 
https://github.com/Invoca/web/pull/24383

### CODE REVIEW

File: lib/active_record/mysql/enum/quoting.rb [hunk 1]
```diff
@@ -8,10 +8,10 @@ module Quoting
       # Quote a symbol as a normal string. This will support quoting of
       # enumerated values.
       def quote(value)
-        if !value.is_a? Symbol
-          __quote_enum(value)
+        if value.is_a? Symbol
+          __quote_enum(value.to_s)
         else
-          ActiveRecord::Base.send(:quote_bound_value, value.to_s)
+          __quote_enum(value)
```
Comment (ColinDKelley) [inline lib/active_record/mysql/enum/quoting.rb]: I'm confused that `__quote_enum` is just an alias for `quote`. Aren't we at risk for infinite recursion in the `else` case?

Oh, I wonder if this is the old "alias method chain" pattern, where `__quote_enum` is aliased to the version of `quote` that was in place before we override it on line 11? If that's true, this pattern has been superseded by `prepend`, so how about if we use that?

```ruby
module QuoteSymbols
  def quote(value)
    super(value.is_a?(Symbol) ? value.to_s : value)
  end
end

ActiveRecord::ConnectionAdapters::Quoting.prepend QuoteSymbols
```

Comment (ColinDKelley) [reply lib/active_record/mysql/enum/quoting.rb]: (But also, I wonder if Rails has started allowing symbols to be quoted? That feels like the sort of thing they would do!)

Comment (ttstarck) [reply lib/active_record/mysql/enum/quoting.rb]: You're 100% correct. I removed this code and all the tests passed. Looking at the source code it looks like they handle Symbols:
https://github.com/rails/rails/blob/v6.0.6.1/activerecord/lib/active_record/connection_adapters/abstract/quoting.rb#L228-L229

Comment (ColinDKelley) [reply lib/active_record/mysql/enum/quoting.rb]: Bingo!

**Review by ColinDKelley — DISMISSED** [review]
Looks great.

Just to confirm: we're not supporting any Rails version now that _doesn't_ natively allow Symbols to be quoted?

**Comment by ttstarck** [comment]
@ColinDKelley yes, that is correct. The lowest Rails version we're supporting is 6.0

**Review by ColinDKelley — APPROVED** [review]

**Pull request merged**